### PR TITLE
Add support for AV1 itags.

### LINKED ISSUE #672: Support AV1 itags
YouTube provides AV1-encoded streams on some videos.

Example video: https://redirect.invidious.io/watch?v=kJQP7kiw5Fk

It seems like it's only available for videos with lots of views, probably because it requires more resources and [their VCU ships don't support AV1 yet](https://blog.youtube/inside-youtube/new-era-video-infrastructure)

AV1 gives us better quality and less bandwith needed.

### PR DESCRIPTION
**Pull request by FireMasterK**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [ ] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [ ] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

I don't really like this approach, since this can include 60 fps streams as well as HDR streams... #488 is the ideal solution, but there's a lot of work to be done on the NewPipe app side which I know nothing about

Closes #672

### CODE REVIEW

**Comment by litetex** [comment]
Tested this on a experimental branch with #488 in the NewPipe app.
https://www.youtube.com/watch?v=kJQP7kiw5Fk worked fine (maybe there was also the old format used by exoplayer idk). 
However videos like https://www.youtube.com/watch?v=aqz-KE-bpKQ failed on higher resolutions (>1080p):
grafik


See also https://github.com/TeamNewPipe/NewPipeExtractor/pull/488#issuecomment-904741877

TL;DR
Would not recommend merging currently

**Comment by FireMasterK** [comment]
Discussion continued on https://github.com/TeamNewPipe/NewPipeExtractor/pull/488#issuecomment-904766848, this behavior happens on high-resolution WEBM and AV1 streams, and even HDR streams and is to be fixed on the App side, and not the extractor.

**Comment by D4v1dH03** [comment]
Unfortunately that means until #488 is merged this won't be merged?

**Comment by FireMasterK** [comment]
Neither of these PRs can be merged currently as there is work to be done in the app to adapt for these changes.

**Comment by DavidH-Tech** [comment]
Android 14 will force the support of software AV1 decoding, even if the hardware does not support. Will you consider supporting it (hopefully) by the time Android 14 is released to general public?

**Review by heroku-miraheze — APPROVED** [review]

**Review by TobiGr — APPROVED** [review]
Sorry, that took me way too long.
